src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    /**
+     * Get score of specific aspect from set (speed | heap usage | serialized size | gc.allocated.rate) after parsing the benchmark result.
+     * @param benchmarkResultFilePath is the path of benchmark result file.
+     * @param keyWord from set (speed | Heap usage | Serialized size | ·gc.alloc.rate) specifies which score will be extracted from the benchmark result.
+     * @return the score of specific aspect in BigDecimal format.
+     * @throws Exception if error occurs when reading Ion Data.
+     */
+    public static BigDecimal getScore(String benchmarkResultFilePath, String keyWord) throws Exception {
+        IonStruct scoreStruct;
+        IonStruct benchmarkResultStruct = readHelper(benchmarkResultFilePath);
+        if (keyWord.equals(SPEED)) {
+            scoreStruct = (IonStruct)benchmarkResultStruct.get(PRIMARY_METRIC);
+        } else {
+            IonStruct secondaryMetricStruct = (IonStruct) benchmarkResultStruct.get(SECONDARY_METRIC);
+            scoreStruct = (IonStruct)secondaryMetricStruct.get(keyWord);
+        }
+        IonValue score = scoreStruct.get(SCORE);
+        if (score.getType().equals(IonType.FLOAT)) {
+            IonFloat scoreFloat = (IonFloat)score;
+            return scoreFloat.bigDecimalValue();
+        } else {
+            IonDecimal scoreDecimal = (IonDecimal)score;
+            return scoreDecimal.bigDecimalValue();
+        }
+    }
+
+    /**
+     * Extract the parameter of specific aspect from set (input | options) out of benchmark result.
+     * @param benchmarkResultFilePath is the path of benchmark result.
+     * @param keyWord specifies which parameter will be returned, from set (input | options).
+     * @return an Ion Value represents the parameter conforms with the keyword.
+     * @throws Exception if error occurs when reading Ion data.
+     */
+    private static IonValue getParameter(String benchmarkResultFilePath, String keyWord) throws Exception{
+        IonStruct benchmarkResultStruct = readHelper(benchmarkResultFilePath);
+        IonStruct parameterStruct = (IonStruct)benchmarkResultStruct.get(PARAMETERS);
+        return parameterStruct.get(keyWord);
+    }
+
+    /**
+     * This is a helper method which create a IonReader for benchmark result and extract the IonStruct which contain parameters or scores information.
+     * @param benchmarkResultFilePath is the path of benchmark result.
+     * @return an Ion Struct which contains information of parameters using during the benchmark process or scores.
+     * @throws Exception if error occurs when reading Ion Data.
+     */
+    private static IonStruct readHelper(String benchmarkResultFilePath) throws Exception {
+        try (IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(benchmarkResultFilePath)))) {
+            reader.next();
+            reader.stepIn();
+            reader.next();
```

Comment:
There should be some checking on the `next()` calls to make sure the reader is positioned on the expected type.

Reply:
> There should be some checking on the `next()` calls to make sure the reader is positioned on the expected type.

Sounds good. Changes will be updated in the next commit.